internal/services/cosmos/cosmosdb_cassandra_datacenter_resource.go

```diff
@@ -276,39 +274,40 @@ func resourceCassandraDatacenterUpdate(d *pluginsdk.ResourceData, meta interface
 }
 
 func resourceCassandraDatacenterDelete(d *pluginsdk.ResourceData, meta interface{}) error {
-	client := meta.(*clients.Client).Cosmos.CassandraDatacentersClient
+	client := meta.(*clients.Client).Cosmos.ManagedCassandraClient
 	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.CassandraDatacenterID(d.Id())
+	id, err := managedcassandras.ParseDataCenterID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	future, err := client.Delete(ctx, id.ResourceGroup, id.CassandraClusterName, id.DataCenterName)
+	future, err := client.CassandraDataCentersDelete(ctx, *id)
 	if err != nil {
-		if !response.WasNotFound(future.Response()) {
+		if !response.WasNotFound(future.HttpResponse) {
 			return fmt.Errorf("deleting %q: %+v", id, err)
 		}
```

Comment:
this is a bug that should be removed:

```suggestion
		return fmt.Errorf("deleting %q: %+v", id, err)
```

Reply:
Updated